architecting-cloud-native-dotnet-apps-for-azure/cloud-native-devops.md

```diff
@@ -5,8 +5,246 @@ ms.date: 06/30/2019
+
+**Azure Repos** - Source code management which supports the venerable Team Foundation Version Control (TFVC) as well as the industry favorite git. Pull requests provide a way to enable social coding by fostring discussion of changes as they are made. 
+
+**Azure Pipelines** - A build and relese management system which supports tight integration with Azure. Builds can be run on a variety of platforms from Windows to Linux to MacOS. Build agents may be provisioned in the cloud or on premise. 
+
+**Azure Test Plans** - No QA person will be left behind with the test management and exploratory testing support offered by the Test Plans Features.
+
+**Azure Artifacts** - An artifact feed which allows companies to create their own, internal, versions of nuget, npm and others. It serves a double purpose of acting as a cache of upstream packages in the event of failure of a centralized repository.
+
+The top-level container in Azure DevOps is known as a Project. The components can be turned on and off for any project. If users want to manage their source code in GitHub but still take advantage of Azure Pipelines then that's perfectly possible. In fact, many open source projects have taken advantages of the [free builds](https://azure.microsoft.com/en-ca/blog/announcing-azure-pipelines-with-unlimited-ci-cd-minutes-for-open-source/) offered by Azure DevOps while keeping their source code on GitHub. Some significant open source project such as [Visual Studio Code](https://code.visualstudio.com/), [yarn](https://yarnpkg.com/en/), [gulp](https://gulpjs.com/) and [NumPy](https://www.numpy.org/) have made the transition. 
+
+Each of these components provides some advantages for Cloud Native Applications but the three most useful are the source control, boards and pipelines.  
+
+## Source Control
+
+Organize the code for a Cloud Native Application can be somewhat challenging. Instead of a single giant application the Cloud Native Applications tend to be made up of a web of smaller applications which talk with one another. As with all things in computing the best arrangement of code remains an open question. There are examples of successful applications using a variety of layouts but two variants seem to have the most popularity. 
+
+Before getting down into the actual source control itself it is probably worth deciding on how many projects are appropriate. Within a single project is support for multiple repositories, and build pipelines. Board are a little more complicated but there too the tasks can easily be assigned to multiple teams within a single project. It is certainly possible to support hundreds, even thousands of developers out of a single Azure DevOps project. Doing so is likely the best approach as it provides a single place for all developer to work out of and reduces the confusion of finding that one application when developers are unsure of even which project in which it resides. 
```

Comment:
"A single project supports multiple repositories and build pipelines."
of even which project in which it resides -> of even in which project it resides